libraries/RC_Channel/RC_Channel.cpp

```diff
@@ -1367,6 +1368,33 @@ bool RC_Channel::do_aux_function(const aux_func_t ch_option, const AuxSwitchPos
            }
         break;
     }
+    
+    case AUX_FUNC::MAG_CAL: {
+        Compass &compass = AP::compass();
+        switch (ch_flag) {
+        case AuxSwitchPos::LOW:
+            if (!hal.util->get_soft_armed()) {
```

Comment:
Any reason its a bad idea to cancel calibration in once armed?

Reply:
Reboots

Reply:
Look like your OK. The `cal_update` stops running.

Reply:
yeah, can drop that qualifier...even if forced arm during cal, and then canceled, won't reboot...